src/cmap/connection.ts

```diff
@@ -680,8 +660,10 @@ export class CryptoConnection extends Connection {
   }
 }
 
-function hasSessionSupport(conn: Connection) {
-  return conn.description.logicalSessionTimeoutMinutes != null;
+/** @public */
```

Comment:
A general question on this file... it looks like this code is non-breaking apart from the unpin server logic and possibly the transient error thing - would it be possible to pull out just the changes here and related files into a separate PR? That way we could ensure that existing functionality, is, in fact, not broken with the new pinning logic? (It's fine if the logic is too intertwined in the other files and the answer is no, I am just trying to think of how to make applying these changes a little safer and easier to review)

Reply:
I think personally I'm happy when the existing tests are running and passing (most notably the transactions spec tests) that the changes here have not broken the existing functionality. Since right now at the time of writing there are only 2 failing tests related to change streams in the main suite I'm pretty confident this is fine.